gnovm/pkg/gnolang/op_call.go

```diff
@@ -154,7 +154,11 @@ func (m *Machine) doOpCall() {
 		}
 		// TODO: some more pt <> pv.Type
 		// reconciliations/conversions necessary.
-		b.Values[i] = pv
+
+		// Make a copy so that a reference to the arguemnt isn't used
+		// in cases where the non-primitive value type is represented
+		// as a pointer, *StructValue, for example.
+		b.Values[i] = pv.Copy(m.Alloc)
```

Comment:
It would be good to know via some repeatable test that the expected number of copies (or the log of copies w/ type copied, one per line) happens and we don't end up doing more copies than needed.

I'm not sure how best to test it though, and I don't think it's a blocker for this. 

seems like a simple filetest should be able to catch the error before 

surprising that this wasn't caught before with a simple

Reply:
I think that the filetest included in the PR is able address this to an extent.